src/dom-matrix-readonly.js

```diff
@@ -0,0 +1,170 @@
+    }
+
+    if (this.is2D && rightMatrix.is2D) {
+      return multiply2DMatrices(this, rightMatrix);
+    }
+
+    var leftMatrix = to3D(this);
+    var rightMatrix = to3D(rightMatrix);
+    return multiply3DMatrices(leftMatrix, rightMatrix);
+  };
+
+  function multiply2DMatrices(left, right) {
+    var result = [];
+    for (var i = 0; i < 3; i++) {
+      for (var j = 0; j < 2; j++) {
+        var value = left._matrix[j]*right._matrix[2*i] +
+            left._matrix[j + 2]*right._matrix[2*i + 1];
+        if (i == 2) {
+          value += left._matrix[2*i + j];
+        }
+        result[j + 2*i] = value;
+      }
+    }
+
+    return new DOMMatrixReadonly([result[0], result[1], result[2], result[3], result[4],
+        result[5]]);
+  };
+
+  function multiply3DMatrices(left, right) {
+    var result = [];
+    var length = 4;
+    for (var i = 0; i < length; i++) {
+      for (var j = 0; j < length; j++) {
+        var value = 0;
+        for (var k = 0; k < length; k++) {
+          value += left._matrix[j + k*length]*right._matrix[i*length + k];
+        }
+        result[i*length + j] = value;
+      }
+    }
+
+    return new DOMMatrixReadonly([result[0], result[1], result[2], result[3], result[4],
+        result[5], result[6], result[7], result[8], result[9], result[10],
+        result[11], result[12], result[13], result[14], result[15]]);
+  };
+
+  function to3D(matrix) {
+    if (!matrix.is2D) {
+      return matrix;
+    }
+
+    var a = matrix._matrix[0];
+    var b = matrix._matrix[1];
+    var c = matrix._matrix[2];
+    var d = matrix._matrix[3];
+    var e = matrix._matrix[4];
+    var f = matrix._matrix[5];
+    // See documentation http://www.w3.org/TR/css3-transforms/#MatrixDefined.
+    return new DOMMatrixReadonly([a, b, 0, 0,
+                      c, d, 0, 0,
+                      0, 0, 1, 0,
+                      e, f, 0, 1]);
+  };
+
+  function matricesApproxEqual(matrix1, matrix2) {
+    var EPSILON = 1e-6;
+    assert.closeTo(matrix1.m11, matrix2.m11, EPSILON);
+    assert.closeTo(matrix1.m12, matrix2.m12, EPSILON);
+    assert.closeTo(matrix1.m13, matrix2.m13, EPSILON);
+    assert.closeTo(matrix1.m14, matrix2.m14, EPSILON);
+    assert.closeTo(matrix1.m21, matrix2.m21, EPSILON);
+    assert.closeTo(matrix1.m22, matrix2.m22, EPSILON);
+    assert.closeTo(matrix1.m23, matrix2.m23, EPSILON);
+    assert.closeTo(matrix1.m24, matrix2.m24, EPSILON);
+    assert.closeTo(matrix1.m31, matrix2.m31, EPSILON);
+    assert.closeTo(matrix1.m32, matrix2.m32, EPSILON);
+    assert.closeTo(matrix1.m33, matrix2.m33, EPSILON);
+    assert.closeTo(matrix1.m34, matrix2.m34, EPSILON);
+    assert.closeTo(matrix1.m41, matrix2.m41, EPSILON);
+    assert.closeTo(matrix1.m42, matrix2.m42, EPSILON);
+    assert.closeTo(matrix1.m43, matrix2.m43, EPSILON);
+    assert.closeTo(matrix1.m44, matrix2.m44, EPSILON);
+  }
+
+  internal.DOMMatrixReadonly = DOMMatrixReadonly;
+  if (TYPED_OM_TESTING) {
```

Comment:
Eh, I forgot that I did eventually add something to the testing scope lol. I created a new internal.testing scope for it, WDYT?